### ISSUE TITLE
Fix dbt project parsing/Airflow DAG rendering for `load_method=LoadMode.CUSTOM`

### ISSUE THREAD

**Issue by tatiana** [astronomer/astronomer-cosmos#561]
As part of testing for the 1.1.2 release (https://github.com/astronomer/astronomer-cosmos/pull/560), we realised that if the DAG uses `load_method=LoadMode.CUSTOM`, all the test nodes are removed from the DAG.

Example DAG, run from the Cosmos repo (dev/dags):
```
import os
from datetime import datetime
from pathlib import Path

from cosmos.constants import LoadMode
from cosmos import DbtDag, ProjectConfig, ProfileConfig, RenderConfig
from cosmos.profiles import PostgresUserPasswordProfileMapping

DEFAULT_DBT_ROOT_PATH = Path(__file__).parent / "dbt"
DBT_ROOT_PATH = Path(os.getenv("DBT_ROOT_PATH", DEFAULT_DBT_ROOT_PATH))

profile_config = ProfileConfig(
    profile_name="default",
    target_name="dev",
    profile_mapping=PostgresUserPasswordProfileMapping(
        conn_id="airflow_db",
        profile_args={"schema": "public"},
    ),
)

basic_cosmos_dag = DbtDag(
    # dbt/cosmos-specific parameters
    project_config=ProjectConfig(
        DBT_ROOT_PATH / "jaffle_shop",
    ),
    render_config=RenderConfig(
        load_method=LoadMode.CUSTOM,
    ),
    profile_config=profile_config,
    operator_args={"install_deps": True},
    # normal dag parameters
    schedule_interval="@daily",
    start_date=datetime(2023, 1, 1),
    catchup=False,
    dag_id="basic_cosmos_dag",
)
```

Rendered DAG with `load_method=LoadMode.DBT_LS`:
![Screenshot 2023-09-27 at 16 32 20](https://github.com/astronomer/astronomer-cosmos/assets/272048/7012fe8c-fb96-4641-bdb4-5006a2eed986)

Same DAG rendered with `load_method=LoadMode.CUSTOM`:
![Screenshot 2023-09-27 at 16 41 33](https://github.com/astronomer/astronomer-cosmos/assets/272048/a4435a0c-2a99-4aa8-9c2f-b6d21ab817e0)

This issue is a bug in the current implementation of Cosmos custom parsing of dbt projects, highlighted since we introduced #543.

**Acceptance criteria**

* [ ] The aforementioned DAG should render test nodes when using `load_method=LoadMode.CUSTOM`

**Comment by raffifu** [comment]
This issue happen because the implementaion of `DbtProject` not properly parsing the test, as far as i know the only way for custom parser to get the test is from reading `*.yml` file and scan if there's a `tests` key on the file. As you can see on the snippet below, the function is only implemented to scan yaml and grab the config. I would like to contribute on this issue

https://github.com/astronomer/astronomer-cosmos/blob/4c8d6b04590db112a904cea0260a4997b36542c4/cosmos/dbt/parser/project.py#L344-L347

**Comment by tatiana** [comment]
Thank you very much, @raffifu ! Your contributions are much appreciated. Please, reach out if you're not able to look into this, and I'll check it out - otherwise, the ticket is yours 🎁

**Comment by tatiana** [comment]
This was released as part of 1.2:
* https://pypi.org/project/astronomer-cosmos/1.2.0/
* https://github.com/astronomer/astronomer-cosmos/releases/tag/astronomer-cosmos-v1.2.0
